contracts/core/modules/RebalanceAuctionModule.sol

```diff
@@ -138,4 +139,61 @@ contract RebalanceAuctionModule is
             _quantity
         );
     }
+
+    /**
+     * If a Rebalancing Set Token Rebalance has failed and been put in Drawdown state, user's
+     * can withdraw their portion of the components collateralizing the Rebalancing Set Token.
+     * This burns the user's portion of the Rebalancing Set Token.
+     *
+     * @param  _rebalancingSetToken    Address of the rebalancing token to withdraw from
+     */
+    function withdrawFromFailedRebalance(
+        address _rebalancingSetToken
+    )
+        external
+        nonReentrant
+    {
+        // Create Rebalancing Set Token instance
+        IRebalancingSetToken rebalancingSetToken = IRebalancingSetToken(_rebalancingSetToken);
+
+        // Make sure the rebalancingSetToken is tracked by Core
+        require(
+            coreInstance.validSets(_rebalancingSetToken),
+            "RebalanceAuctionModule.bid: Invalid or disabled SetToken address"
+        );
+
+        // Get combinedTokenArray from RebalancingSetToken
```

Comment:
Did you want to check the state here before doing this?

Reply:
There's a state check that occurs in the burn function...so didn't want to get duplicative